contracts/margin/external/exchangewrappers/ZeroExV2ExchangeWrapper.sol

```diff
@@ -0,0 +1,295 @@
+        IExchange v2Exchange = IExchange(ZERO_EX_EXCHANGE);
+        OrderInfo memory orderInfo = v2Exchange.getOrderInfo(order);
+
+        if (orderInfo.orderStatus != uint8(OrderStatus.FILLABLE)) {
+            return 0;
+        }
+
+        uint256 remainingTakerAssetAmount =
+            order.takerAssetAmount.sub(orderInfo.orderTakerAssetFilledAmount);
+
+        return MathHelpers.getPartialAmount(
+            remainingTakerAssetAmount,
+            order.takerAssetAmount,
+            order.makerAssetAmount
+        );
+    }
+
+    // ============ Private Functions ============
+
+    function ensureAllowance(
+        address token,
+        address spender,
+        uint256 requiredAmount
+    )
+        private
+    {
+        if (token.allowance(address(this), spender) >= requiredAmount) {
+            return;
+        }
+
+        token.approve(
+            spender,
+            MathHelpers.maxUint256()
+        );
+    }
+
+    function transferTakerFee(
+        Order memory order,
+        address tradeOriginator,
+        uint256 requestedFillAmount
+    )
+        private
+    {
+        uint256 takerFee = MathHelpers.getPartialAmount(
+            requestedFillAmount,
+            order.takerAssetAmount,
+            order.takerFee
+        );
+
+        if (takerFee == 0) {
+            return;
+        }
+
+        require(
+            TRUSTED_MSG_SENDER[msg.sender],
+            "ZeroExV2ExchangeWrapper#transferTakerFee: Only trusted senders can dictate the fee payer"
+        );
+
+        ZRX.transferFrom(
+            tradeOriginator,
+            address(this),
+            takerFee
+        );
+    }
+
+    function parseSignature(
+        bytes orderData
+    )
+        private
+        pure
+        returns (bytes memory)
+    {
+        bytes memory signature = new bytes(66);
+
+        /* solium-disable-next-line security/no-inline-assembly */
+        assembly {
+            mstore(add(signature, 32), mload(add(orderData, 352)))  // first 32 bytes
+            mstore(add(signature, 64), mload(add(orderData, 384)))  // next 32 bytes
+            mstore(add(signature, 66), mload(add(orderData, 386)))  // last 2 bytes
+        }
+
+        return signature;
+    }
+
+    function parseOrder(
+        bytes orderData,
+        address makerToken,
+        address takerToken
+    )
+        private
+        pure
+        returns (Order memory)
+    {
+        Order memory order;
+
+        /* solium-disable-next-line security/no-inline-assembly */
+        assembly {
+            mstore(order,           mload(add(orderData, 32)))  // maker
+            mstore(add(order, 32),  mload(add(orderData, 64)))  // taker
+            mstore(add(order, 64),  mload(add(orderData, 96)))  // feeRecipient
+            mstore(add(order, 96),  mload(add(orderData, 128))) // sender
+            mstore(add(order, 128), mload(add(orderData, 160))) // makerTokenAmount
```

Comment:
update these comments to be the 0x v2 terminology

this should be `makerAssetAmount`

Reply:
See https://0xproject.com/docs/0xjs#types-SignedOrder